rust/hyperlane-base/src/settings/loader/mod.rs

```diff
@@ -72,10 +75,11 @@ where
                 .separator("_")
                 .source(Some(filtered_env)),
         )
+        .add_source(CommandLineArguments::default().separator("."))
```

Comment:
LMK if you think we should use something else such as `-`.

This currently looks like `--chains.ethereum.conenction.url="https://example.com"` but we could also do something like `--chains-ethereum-conenction-url="https://example.com"`.